frontend/src/components/GroupList.tsx

```diff
@@ -88,6 +85,23 @@ export default function GroupList(props: { session: Session | null }) {
   useEffect(() => {
     refreshGroups();
   }, [isOpen]);
+  useEffect(() => {
```

Comment:
This doesn't need to be a useEffect, since it has no side effects (making API requests, modifying DOM elements).

This can be a pure function instead.

```
function getSubscriptionCost() {
  let totalCost = 0;
  ...
}

subscriptionCost = getSubscriptionCost()
```

It's called a "computed property", since it's calculated it using the current groups & userData and does not modify any other state (thus having no side effect).

Reply:
fixed